lib/models/release.ts

```diff
@@ -92,47 +95,87 @@ const getReleaseModel = function (
 	 * });
 	 *
 	 * @example
+	 * balena.models.release.get({raw_version: '0.0.0', application: 456}).then(function(release) {
+	 * 	console.log(release);
+	 * });
+	 *
+	 * @example
 	 * balena.models.release.get(123, function(error, release) {
 	 * 	if (error) throw error;
 	 * 	console.log(release);
 	 * });
 	 */
 	async function get(
-		commitOrId: string | number,
+		commitOrIdOrRawVersion:
+			| string
+			| number
+			| { rawVersion: string; application: number | string },
 		options: BalenaSdk.PineOptions<BalenaSdk.Release> = {},
 	): Promise<BalenaSdk.Release> {
-		if (commitOrId == null) {
-			throw new errors.BalenaReleaseNotFound(commitOrId);
+		if (commitOrIdOrRawVersion == null) {
+			throw new errors.BalenaReleaseNotFound(commitOrIdOrRawVersion);
 		}
 
-		if (isId(commitOrId)) {
+		if (isId(commitOrIdOrRawVersion)) {
 			const release = await pine.get({
 				resource: 'release',
-				id: commitOrId,
+				id: commitOrIdOrRawVersion,
 				options: mergePineOptions({}, options),
 			});
 			if (release == null) {
-				throw new errors.BalenaReleaseNotFound(commitOrId);
+				throw new errors.BalenaReleaseNotFound(commitOrIdOrRawVersion);
 			}
 			return release;
+		} else if (typeof commitOrIdOrRawVersion === 'object') {
+			const { rawVersion, application } = commitOrIdOrRawVersion;
+			if (!bSemver.valid(commitOrIdOrRawVersion.rawVersion)) {
+				throw new errors.BalenaReleaseNotFound(rawVersion); // TODO: make balena-errors accept an object for this
+			}
+			const { id } = await applicationModel().get(application, {
+				$select: 'id',
+			});
+			const releases = await pine.get({
+				resource: 'release',
+				options: mergePineOptions(
+					{
+						$filter: {
+							raw_version: rawVersion,
+							belongs_to__application: id,
+						},
+					},
+					options,
+				),
+			});
+			if (releases.length === 0) {
+				throw new errors.BalenaReleaseNotFound(
+					JSON.stringify(commitOrIdOrRawVersion),
+				); // TODO
```

Comment:
is this TODO necessary ? if so, could we just add a small description to make it more clear please ?

Reply:
This is a TODO I will handle before finish the PR. The gist is that `errors.BalenaReleaseNotFound` is not made to accept the object that can be passed for rawVersion, so I would like to update it to accept such objects and treat them the way the `getFieldLabel` function I added to the `utils` file works